django-stubs/db/migrations/migration.pyi

```diff
@@ -1,17 +1,20 @@
 from collections.abc import Sequence
+from typing import ClassVar
 
 from django.db.backends.base.schema import BaseDatabaseSchemaEditor
 from django.db.migrations.operations.base import Operation
 from django.db.migrations.state import ProjectState
 from typing_extensions import Self
 
 class Migration:
-    operations: Sequence[Operation]
-    dependencies: list[tuple[str, str]]
-    run_before: list[tuple[str, str]]
-    replaces: list[tuple[str, str]]
-    initial: bool | None
-    atomic: bool
+    # Django copies these 4 attributes from the class to the instance,
+    # but they're practically used as ClassVar
+    operations: ClassVar[Sequence[Operation]]
```

Comment:
Do we have real evidence that instance variables *won't work*?

Reply:
Yes. If you remove the `ClassVar` annotation on `operations`, the `test_classvar.yml::explicit` test case fails with:
>  error: Cannot override instance variable (previously declared on base class "Migration") with class variable (diff).

These attributes are nearly always defined by downstreams as `ClassVar`, so if we want to support the correct annotations (which tools like Ruff require), the upstream needs to match.

Of course, downstreams which do not explicitly annotate these attributes will be unaffected, as the `test_classvar.yml::implicit` test case shows.

Reply:
Will users experience any bugs if they use these variables as instance attributes, not class variables?

A common problem that exists and that is solved by `ClassVar` is this pattern: `MyClass.my_var` - in this case we need to have `my_var: ClassVar`, otherwise it will fail with `AttributeError`

Reply:
> Will users experience any bugs if they use these variables as instance attributes, not class variables?

You are allowed to make use of class variables "as instance attributes", i.e.:

```python
class MyMigration(Migration):
    def some_method(self):
        if self.operations == ...:  # Fine
            ...
        self.operations = ...  # Error
```

If this is what you are asking